web/src/app/ee/admin/groups/[groupId]/GroupDisplay.tsx

```diff
@@ -95,30 +101,61 @@ const UserRoleDropdown = ({
     }
   };
 
+  const handleChange = (value: string) => {
+    if (value === UserRole.BASIC && user.id === currentUser?.id) {
+      setPendingRoleChange(value);
+      setShowDemoteConfirm(true);
+    } else {
+      applyRoleChange(value);
+    }
+  };
+
   const isEditable =
-    (user.role === UserRole.BASIC || user.role === UserRole.CURATOR) && isAdmin;
-
-  if (isEditable) {
-    return (
-      <div className="w-40">
-        <Select
-          value={localRole}
-          onValueChange={handleChange}
-          disabled={isSettingRole}
-        >
-          <SelectTrigger>
-            <SelectValue placeholder="Select role" />
-          </SelectTrigger>
-          <SelectContent>
-            <SelectItem value={UserRole.BASIC}>Basic</SelectItem>
-            <SelectItem value={UserRole.CURATOR}>Curator</SelectItem>
-          </SelectContent>
-        </Select>
-      </div>
-    );
-  } else {
-    return <div>{USER_ROLE_LABELS[localRole]}</div>;
-  }
+    user.role === UserRole.BASIC || user.role === UserRole.CURATOR;
+
+  return (
+    <>
+      {/* Confirmation modal - only shown when users try to demote themselves */}
+      {showDemoteConfirm && pendingRoleChange && (
+        <GenericConfirmModal
```

Comment:
Would refresh user here or an alternative- user should likely be redirected to chat page